eng/targets/ResolveReferences.targets

```diff
@@ -119,8 +117,8 @@
 
       <!-- Ensure only content asset are consumed from .Sources packages -->
       <Reference>
-        <IncludeAssets Condition="'%(SharedSources)' == 'true'">ContentFiles</IncludeAssets>
-        <PrivateAssets Condition="'%(SharedSources)' == 'true'">All</PrivateAssets>
+        <IncludeAssets Condition="$([System.String]::new('%(Identity)').EndsWith('.Sources'))">ContentFiles</IncludeAssets>
```

Comment:
What about setting `SharedSources` metadata conditionally on `Condition="$([System.String]::new('%(Identity)').EndsWith('.Sources'))` and then using it in all other places.

Reply:
that would have worked too.

Reply:
Would allow setting `SharedSources` for references that do not end in `.Sources` and keep the logic in one place.

Reply:
updated.